architecture.md

```diff
@@ -8,14 +8,14 @@ Below are the major components deployed with the [Kubecost helm chart](http://do
     b. Cost-model -- provides cost allocation calculations and metrics, both reads and writes to Prometheus  
 2. **Prometheus**  
     a. Prometheus server -- time-series data store for cost & health metrics  
-    b. Kube-state-metrics -- provides Kubernetes API metrics, e.g. resource requests  
-    c. Node-exporter -- provides node-level utilization metrics for right-sizing recommendations and cluster utilization  
```

Comment:
@srpomeroy I believe NE still used for autoscaling cluster right-sizing? Otherwise this LGTM!

Reply:
@dwbrown2 There is much confusion around NE and where it is actually needed. We recently added a resource tuning guide to our docs. [There is a section on NE](https://guide.kubecost.com/hc/en-us/articles/6446286863383-Tuning-Resource-Consumption#disable-or-stop-scraping-node-exporter) that claims disabling it only impacts cluster health. Is that not accurate?

Reply:
@srpomeroy That doesn't appear accurate based on just an issue I looked into the other day: https://github.com/kubecost/cost-analyzer-frontend/issues/1023#issuecomment-1231989077

These look like node-exporter metrics yeah?
```
query=sum(node_memory_MemTotal_bytes - node_memory_MemFree_bytes - node_memory_Cached_bytes - node_memory_Buffers_bytes) / 1024 / 1024 / 1024
```

I have brought this up multiple times that there are likely a handful of node exporter queries that could effect more parts of the application. They are almost all legacy features and are somewhat difficult to track down until they pop up in bug tickets.

Reply:
Here are all the areas that I'm aware of:

* Reserved Instance Recommendations in Savings
* Various Kubecost Grafana dashboards
* Several health alerts

None of these are critical to core Kubecost. Are we missing any?

Before redoing overview, we closely tested without NE to confirm and came up with these: https://guide.kubecost.com/hc/en-us/articles/4407595942679-Kubecost-Diagnostics#node-exporter-metrics-available

They should be updated. Let me know if I can help with this. Also, cc @bstuder99 for docs!